app/views/charges/_form.html.erb

```diff
@@ -0,0 +1,13 @@
+<% if current_user.premium? %>
+  <%= render partial: "charges/downgrade" %>
+<% else %>
+  <%= form_tag charges_path do %>
+    <h4>Upgrade To Premium</h4>
+    <script class='stripe-button' src="https://checkout.stripe.com/checkout.js"
+                                  data-key="<%= @stripe_btn_data[:key] %>"
+                                  data-amount=<%= @stripe_btn_data[:amount] %>
```

Comment:
@bjblock - When I use this partial in `charges#new`, it works great. But When I'm trying to use it within my `user#show` page, I'm getting the following error:
```
NoMethodError in Users#show
undefined method `[]' for nil:NilClass

data-key="<%= @stripe_btn_data[:key] %>
```

Normally this would just be an error because something is returning nil right?